packages/storybook/src/builders/storybook-build/storybook-build.impl.spec.ts

```diff
@@ -0,0 +1,5 @@
+describe('storybook builer', () => {
+  it('should have a test', () => {
```

Comment:
Are there any more unit tests that would be useful here?

Reply:
I can have a look at this, but my feeling is that this will be equal to the e2e test. If you have any suggestions on how to tackle this, please let me know.

Reply:
I think testing the webpack config which gets used here is useful. Maybe look at the way the web and node build builders are tested :wink: .

Reply:
@FrozenPandaz I have tried to work an a similar way like the web builder spec, but I was able to get a successful run. It start running, but on the part it starts compiling it stats to go wrong
```
 clean outputDir..
 => Copying prebuild dll's..
 => Building manager..
 => Loading manager config..
 => Loading presets
 => Compiling manager..

ERR! => Failed to build the manager
ERR!   TypeError: Cannot read property '_location' of null
ERR!   
ERR!   - Window.js:223 Window.get location [as location]
ERR!     [nx]/[jest-environment-jsdom]/[jsdom]/lib/jsdom/browser/Window.js:223:79
ERR!   
ERR!   - lodash.js:2392 arrayLikeKeys
ERR!     [nx]/[lodash]/lodash.js:2392:16
ERR!   
ERR!   - lodash.js:13334 keysIn
ERR!     [nx]/[lodash]/lodash.js:13334:36
ERR!   
ERR!   - lodash.js:12639 
ERR!     [nx]/[lodash]/lodash.js:12639:26
ERR!   
ERR!   - lodash.js:4853 Function.<anonymous>
ERR!     [nx]/[lodash]/lodash.js:4853:13
ERR!   
ERR!   - lodash.js:475 apply
ERR!     [nx]/[lodash]/lodash.js:475:27
ERR!   
ERR!   - lodash.js:6563 Function.<anonymous>
ERR!     [nx]/[lodash]/lodash.js:6563:16
ERR!   
ERR!   - index.js:336 HtmlWebpackPlugin.evaluateCompilationResult
ERR!     [nx]/[core]/[html-webpack-plugin]/index.js:336:42
ERR!   
ERR!   - index.js:258 
ERR!     [nx]/[core]/[html-webpack-plugin]/index.js:258:25
ERR!   
ERR!   - async Promise.all
ERR!   
ERR! 
(node:2071) UnhandledPromiseRejectionWarning: Error: 
    at Env.it (/mnt/d/GIT/nx/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:91:24)
    at it (/mnt/d/GIT/nx/node_modules/jest-jasmine2/build/jasmine/jasmineLight.js:93:21)
    at Suite.<anonymous> (/mnt/d/GIT/nx/build/packages/storybook/src/builders/build-storybook/build-storybook.impl.spec.js:20:5)
    at addSpecsToSuite (/mnt/d/GIT/nx/node_modules/jest-jasmine2/build/jasmine/Env.js:496:51)
    at Env.describe (/mnt/d/GIT/nx/node_modules/jest-jasmine2/build/jasmine/Env.js:466:11)
    at describe (/mnt/d/GIT/nx/node_modules/jest-jasmine2/build/jasmine/jasmineLight.js:81:18)
    at Object.<anonymous> (/mnt/d/GIT/nx/build/packages/storybook/src/builders/build-storybook/build-storybook.impl.spec.js:15:1)
    at Runtime._execModule (/mnt/d/GIT/nx/node_modules/jest-runtime/build/index.js:867:68)
    at Runtime._loadModule (/mnt/d/GIT/nx/node_modules/jest-runtime/build/index.js:577:12)
    at Runtime.requireModule (/mnt/d/GIT/nx/node_modules/jest-runtime/build/index.js:433:10)
(node:2071) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:2071) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
error Command failed with exit code 1.
```

Reply:
@kristofdegrave I can take a look at the unit test code, but I don't think you pushed it up

Reply:
@isaacplmann tnx! I've pushed the changes now.

Reply:
In `storybook/src/utils/testing.ts`

```
import { createEmptyWorkspace } from '@nrwl/workspace/src/utils/testing-utils';
```

I found this issue by running the test in debug mode:

```
node --inspect-brk ./node_modules/jest/bin/jest.js build/packages/storybook/src/builders/build-storybook/build-storybook.impl.spec.js
```

Reply:
@isaacplmann tnx for the hint! Unit test now seem to work :)

Reply:
@isaacplmann it seems the tests are running local, but it keeps giving an issue on the buildserver

`Unhandled rejection TypeError: setTimeout(...).unref is not a function`